### ISSUE TITLE
Does transduce take its arguments in the wrong order?

### ISSUE THREAD

**Issue by TheLudd** [ramda/ramda#2123]
I realized something yesterday about `transduce` that has bothered me for a long time. Every time I use it I pretty much make a custom transducer function but the accumulator function and the initial argument pretty much only varies in two ways. 

 * Case 1: I build up a list of elements where xf is `flip(append)` and acc is `[]`
 * Case 2: I build up a sum where xf is `add` and acc is `0`

Because of this, it seems a better argument order would be `transduce(xf, acc, transducer, list)`. With this I could create 
 * `transduceToList = transduce(flip(append), [])` 
 * `transduceToSum = transduce(add, 0)` 

I suspect that other uses of `transduce` would follow the same pattern.

I realize that `transduce` is not a ramda invention and that it probably follows a format from somewhere else. I just wanted to bring this up for discussion.

**Comment by CrossEye** [comment]
This makes sense to me.  I don't think the prior art should dictate our parameter order.

Of course this would be a major breaking change.

**Comment by TheLudd** [comment]
@CrossEye glad we agree at least =)

**Comment by kedashoe** [comment]
I like it. My only thought about a PR here is that the general consensus seems to be that we should rip transducers out of the core, so I'm not sure this is worth a PR since it is a breaking change?

**Comment by CrossEye** [comment]
@kedashoe: We'll see if I'm really back on doing some real Ramda work, but if so, I agree.  I think we want this done with transducers wherever they land, but it's not clear yet where that will be.

**Comment by buzzdecafe** [comment]
> the general consensus seems to be that we should rip transducers out of the core

Yes, but I have been having trouble imagining how to do so in such a way that they can be transparently plugged in.

**Comment by franciscotln** [comment]
I'm having problem in understanding something kinda related to "order".
I'm using R.into with a composed transducer but the order of the functions calls inside R.compose is the opposite, like pipe, and vice-versa: using R.pipe the functions calls are called from left to right.
https://goo.gl/ZuKUvN

Is this the expected behaviour?

**Comment by CrossEye** [comment]
Yes, this is the expected behavior.  There are several good articles, mostly Clojure related:

  - http://gmorpheme.net/why-transducers-compose-backwards.html
  - http://isaaccambron.com/blog/2014/12/13/transducer-composition.html
